validation-test/Sema/type_checker_perf/fast/sr139.swift

```diff
@@ -3,6 +3,5 @@
 
 // SR-139:
 // Infinite recursion parsing bitwise operators
-// expected-error@+1 {{unable to type-check this expression in reasonable time}}
 let x = UInt32(0x1FF)&0xFF << 24 | UInt32(0x1FF)&0xFF << 16 | UInt32(0x1FF)&0xFF << 8 | (UInt32(0x1FF)&0xFF)
```

Comment:
It is still slow or have these changes make this expression fast?

Reply:
@xedin It's much faster with this change, previously it took ~8 seconds, now ~0.05 seconds (and ~4 seconds for `-disable-constraint-solver-performance-hacks`). It looks like we're now visit far fewer scopes.

Reply:
That's pretty interesting, do you know whether it's because we now active constraints which fail faster?

Reply:
I'm trying to ask whether this is a consequence of us activating a set of constraints which we didn't before that help components fail faster?

Reply:
@xedin Yes, that appears to be the case. In fact it looks like it was due to a bug with the old implementation where if a type variable in the equivalence class was visited as a constraint adjacency, it wouldn't then have its fixed bindings visited, missing some constraints.

Reply:
Great!